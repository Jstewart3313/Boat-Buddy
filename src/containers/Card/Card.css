.card {
  align-items: center;
  background-color: #FFFFFF;
  border-bottom: 2px solid #F5F5F5;
  display: flex;
  justify-content: space-between;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 10px;
  padding-top: 5px;
  width: 100%;
}

.disabled {
  align-items: center;
  background-color: #F5F5F5;
  border-bottom: 2px solid #F5F5F5;
  border-radius: 20px;
  display: flex;
  justify-content: space-between;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 10px;
  padding-top: 5px;
  width: 100%;
}

.done {
  border: none;
  border: 1px solid #F3A547;
  border-radius: 10px;
  color: #666666;
  cursor: pointer;
}

.done:hover {
  background-color: #3376E8;
  border-color: #3376E8;
  color: #FFFFFF
}

.done-disabled{
  background-color: lightgrey;
  border: none;
  border: 1px solid #F3A547;
  border-radius: 10px;
  color: #666666;
}

.link-disabled {
  color: #666666;
  line-height: 5px;
  text-decoration: none;
}

.link {
  color: #3376E8;
  line-height: 5px;
  text-decoration: none;
}

.link:hover {
  text-decoration: underline;
}