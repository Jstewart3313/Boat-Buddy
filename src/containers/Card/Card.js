import React, { Component } from "react";
import { connect } from "react-redux";
import "./Card.css";
import { incrementMusicAction } from "../../actionCreators/incrementMusicAction";
import { incrementMovieAction } from "../../actionCreators/incrementMovieAction";
import { incrementTvShowAction } from "../../actionCreators/incrementTvShowAction";

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      done: false
    };
  }
  handleClick = event => {
    const { name } = event.target;
    if (this.state.done === false) {
      this.setState({ done: true });
    } else {
      this.setState({ done: false });
    }
    if (name === "movie" && !this.state.done) {
      this.props.incrementMovie(name);
    } else if (name === "show" && !this.state.done) {
      this.props.incrementTvShow(name);
    } else if( name === "music" && !this.state.done) {
      this.props.incrementMusic(name);
    }
  };
  render() {
    return (
      <div className={this.state.done ? "disabled" : "card"}>
        <a className={this.state.done ? 'link-disabled' : "link"}
           href={this.props.Link}>
          {this.props.name}
        </a>
        <button
          disabled={this.state.done}
          className={this.state.done ? 'done-disabled' : "done"}
          onClick={this.handleClick}
          name={this.props.type}
        >
          DONE
        </button>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  incrementMovie: name => dispatch(incrementMovieAction(name)),
  incrementMusic: name => dispatch(incrementMusicAction(name)),
  incrementTvShow: name => dispatch(incrementTvShowAction(name))
});

export default connect(
  null,
  mapDispatchToProps
)(Card);
