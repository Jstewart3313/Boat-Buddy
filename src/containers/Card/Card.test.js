import React from "react";
import { shallow } from "enzyme";
import { Card } from "./Card";

describe("Card", () => {
  it("should match the snapshot", () => {
    let wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
});
