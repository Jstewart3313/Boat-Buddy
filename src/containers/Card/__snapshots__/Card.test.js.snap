// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div
  className="card"
>
  <a
    className="link"
  />
  <button
    className="done"
    onClick={[Function]}
  >
    DONE
  </button>
</div>
`;
